lib/tasks/production.rake

```diff
@@ -0,0 +1,152 @@
+    system("kubectl delete pod #{pod_from_deployment(deployment)}")
+  end
+
+  def self.console(deployment, cmd = "/bin/bash")
+    system("kubectl exec --stdin --tty #{pod_from_deployment(deployment)} -- #{cmd}")
+  end
+
+  def self.update_deployment_image(deployment, image)
+    system("kubectl get deployment #{deployment} -o json | jq '.spec.template.spec.containers[].image = \"#{image}\"' | kubectl replace -f -")
+  end
+end
+
+# rubocop:disable Style/StderrPuts
+
+namespace :production do
+  task :set_context do # rubocop:disable Rails/RakeEnvironment
+    context   = "do-nyc1-miq-prod"
+    name      = "miq-prod"
+    namespace = "bot"
+
+    unless Kubernetes.available?
+      $stderr.puts "ERROR: You must install kubectl command"
+      exit 1
+    end
+
+    unless Kubernetes.context?(context)
+      puts "Configuring Kubernetes context..."
+
+      config = Kubernetes.config_dir.join("#{name}-kubeconfig.yaml")
+      unless config.exist?
+        $stderr.puts
+        $stderr.puts "ERROR: Cannot find the context config file"
+        $stderr.puts
+        $stderr.puts "1. Go to digitalocean.com => Kubernetes => #{name} => Download Config File"
+        $stderr.puts "2. Place the file in #{Kubernetes.config_dir}"
+        exit 1
+      end
+
+      Kubernetes.merge_config(config)
+      config.delete
+
+      puts "Configuring Kubernetes context...Complete"
+    end
+
+    exit 1 unless Kubernetes.use_context(context) && Kubernetes.use_namespace(namespace)
+    puts
+  end
+
+  desc "Edit the bot settings in production"
+  task :edit_settings => :set_context do
+    exit 1 unless Kubernetes.edit_config_map("bot-settings")
+
+    puts "Restarting the queue-worker pod..."
+    exit 1 unless Kubernetes.restart_deployment_pods("queue-worker")
+    puts "Restarting the queue-worker pod...Complete"
+  end
+
+  desc "Open a console in production"
+  task :console => :set_context do
+    exit 1 unless Kubernetes.console("ui", "/bin/bash -c \"source container-assets/container_env; bash\"")
+  end
+end
+
+desc "Release a new version to production"
+task :release, [:version, :remote] => "production:set_context" do |_t, args|
+  version = args[:version]
+  if version.nil?
+    $stderr.puts "ERROR: must specify the version number to deploy"
+    exit 1
+  end
+  version = "v#{version}" unless version.start_with?("v")
+
+  remote = args[:remote] || "upstream"
+
+  puts "Deploying version #{version}..."
+
+  puts
+  puts "Tagging repository..."
+  if `git tag --list #{version}`.chomp.empty?
+    exit 1 unless system("git tag #{version}")
+  elsif `git tag --points-at HEAD`.chomp != version
+    $stderr.puts "ERROR: Already tagged '#{version}', but you are not on that commit"
+    exit 1
+  else
+    puts "Already tagged '#{version}'"
+  end
+  exit 1 unless system("git push #{remote} #{version}")
+
+  image = "manageiq/miq_bot:#{version}"
+
+  puts
+  puts "Building docker image..."
+  exit 1 unless system("docker build . -t #{image}")
```

Comment:
Not sure if it's preferable to do a `--no-cache` here or not.

Reply:
Yes, I usually do `--no-cache`

Reply:
yeah it actually requires it, otherwise it doesn't detect a code change and uses the cache for the entire thing.

Reply:
Added